File: Java_ChildrenCompetition/src/main/java/service/TestParticipantRelationService.java
package service;

import model.TestParticipantRelation;
import model.Tuple;
import repository.ITestParticipantRelationRepository;

public class TestParticipantRelationService implements ITestParticipantRelationService {
    public final ITestParticipantRelationRepository testParticipantRelationRepository;

    public TestParticipantRelationService(ITestParticipantRelationRepository testParticipantRelationRepository) {
        this.testParticipantRelationRepository = testParticipantRelationRepository;
    }

    @Override
    public void save(int idTest, int idParticipant) {
        Tuple<Integer, Integer> id = new Tuple<>(idTest, idParticipant);
        TestParticipantRelation testParticipantRelation = new TestParticipantRelation(id);
        this.testParticipantRelationRepository.save(testParticipantRelation);
    }
}

File: CSharp_ChildrenCompetitionGUI/CSharp_ChildrenCompetitionGUI/CSharp_ChildrenCompetitionGUI/service/TestParticipantRelationService.cs
using System;
using CSharp_ChildrenCompetitionGUI.model;
using CSharp_ChildrenCompetitionGUI.repository;

namespace CSharp_ChildrenCompetitionGUI.service
{
    public class TestParticipantRelationService : ITestParticipantRelationService<Tuple<int, int>, TestParticipantRelation>
    {
        public ITestParticipantRelationRepository<Tuple<int, int>, TestParticipantRelation> testParticipantRelationRepository;

        public TestParticipantRelationService(ITestParticipantRelationRepository<Tuple<int, int>, TestParticipantRelation> testParticipantRelationRepository)
        {
            this.testParticipantRelationRepository = testParticipantRelationRepository;
        }

        public void save(int idTest, int idParticipant)
        {
            // throw new NotImplementedException();
            Tuple<int, int> id = new Tuple<int, int>(idTest, idParticipant);
            TestParticipantRelation testParticipantRelation = new TestParticipantRelation(id);
            this.testParticipantRelationRepository.save(testParticipantRelation);
        }
    }
}